### ISSUE TITLE
`match_wild_err_arm` triggers weirdly

### ISSUE THREAD

**Issue by Lokathor** [rust-lang/rust-clippy#5024]
If you match on `Err(_)` then it triggers the `match_wild_err_arm` lint and clippy tells you to match on each error individually, but if you match on `Err(_e)` it doesn't trigger at all, even though that's ignoring all possible errors just the same.

Code that brought this up
```rust
  #[inline]
  #[must_use]
  pub fn from_array_len(data: A, len: usize) -> Self {
    match Self::try_from_array_len(data, len) {
      Ok(out) => out,
      Err(_) => {
        panic!("ArrayishVec: length {} exceeds capacity {}!", len, A::CAPACITY)
      }
    }
  }
```

* This should probably only trigger on enum errors, not struct errors.
* Using a name other than `_` will still match on everything, which is what this lint theoretically doesn't link, so alternate names should also still trigger the lint

**Comment by JohnTitor** [comment]
Actually, the lint description says it checks for arm which matches all errors with `Err(_)` so the identifiers like `_e` seem unintended.
But it might be nice that this lint also checks the identifiers which have `_` as a prefix.

**Comment by flip1995** [comment]
We rarely have issues demanding that a lint should lint **more**, so I like this :smile: 

When linting on `_e` and similar, a check is required, that `_e` isn't used in the match arm though.
